### PR TITLE
Work on updating specs (and a few useful features)

### PR DESCRIPTION
**Pull request by ahmadine**
This is a part of my ongoing work on modernizing the codebase.

On a sidenote, I think that the return of promises may at least be broken in a browser runner - I can't trigger a failure in async code. Unfortunately I lack the ability to debug this.

### CODE REVIEW

**Comment by ahmadine** [comment]
I think I can do nothing more about BrowserStack tests. There's probably "BS_AUTHKEY" environment variable missing, that's what I understand from this error message:

*** Error: Atleast one argument is required!

I tested the BrowserStackLocal application locally and it only triggers this message when there's no key provided.

**Comment by ahmadine** [comment]
I know I could have broken it all up to individual pull requests, but I hope it doesn't make your work harder (it's much easier for me that way). I try to describe the commits as well as I can.

**Comment by elia** [comment]
> I know I could have broken it all up to individual pull requests, but I hope it doesn't make your work harder (it's much easier for me that way). I try to describe the commits as well as I can.

No problem, I'll review the commit one by one (as I usually do) it will just take time. I think I'll do multiple reviews so you can get early feedback 👍

File: opal/browser/dom/element.rb [hunk 1]
```diff
@@ -351,7 +351,17 @@ def scroll
...
+        xpath(selector)
+      rescue Exception
+        []
+      end.to_a.concat(
+        begin
+          css(selector)
+        rescue Exception
+          []
+        end.to_a
+      )
```
Comment (elia) [inline opal/browser/dom/element.rb]: as above, the `rescue` should go inside `xpath/css`

File: opal/browser/event/base.rb [hunk 2]
```diff
@@ -270,6 +270,37 @@ def attach!(*)
...
+    #   @return [Delegate]
+    def one (name, selector = nil, &block)
+      raise ArgumentError, 'no block has been given' unless block
+
+      cb = on name, selector do |*args|
+        out = block.call(*args)
+        cb.off
+        out
+      end
+    end
```
Comment (elia) [inline opal/browser/event/base.rb]: 💯

**Comment by ahmadine** [comment]
Thank you for this review. I fixed all the problems you mentioned.

**Pull request merged**